ironic/drivers/modules/drac/management.py

```diff
@@ -302,14 +315,182 @@ def set_boot_device(node, device, persistent=False):
+        for manager in system.managers:
+            try:
+                oem_manager = manager.get_oem_extension('Dell')
+            except sushy.exceptions.OEMExtensionNotFoundError as e:
+                error_msg = (_("Search for Sushy OEM extension package "
+                            "'sushy-oem-idrac' failed for node %(node)s. "
+                            "Ensure it is installed. Error: %(error)s") %
+                            {'node': task.node.uuid, 'error': e})
+                LOG.error(error_msg)
+                raise exception.RedfishError(error=error_msg)
+            try:
+                delete_job_response = oem_manager.clear_job_queue(
+                                                job_ids=['JID_CLEARALL'])
+            except sushy.exceptions.SushyError as e:
+                LOG.debug("Sushy OEM extension Python package "
+                          "sushy-oem-idrac failed to clear job queue"
+                          "for %(node)s, Error : %(error)s" %
+                          {'node':task.node.uuid, 'error':e})
+
+        if delete_job_response.status_code == _JOB_RESPONSE_CODE:
+            LOG.info("Job queue cleared for node %(node)s via OEM" %
+                    {'node': task.node.uuid})
+        else:
+            LOG.error("Failed to clear job queue, node : %(node)s " %
+                     {'node': task.node.uuid})
+        return delete_job_response
+
+    @METRICS.timer('DracRedfishManagement.reset_idrac')
+    @base.clean_step(priority=0)
+    def reset_idrac(self, task):
+        """Reset the iDRAC.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: DracOperationError on an error from python-dracclient.
+        """
+        system = redfish_utils.get_system(task.node)
+        for manager in system.managers:
+            try:
+                oem_manager = manager.get_oem_extension('Dell')
+            except sushy.exceptions.OEMExtensionNotFoundError as e:
+                error_msg = (_("Search for Sushy OEM extension package "
+                           "'sushy-oem-idrac' failed for node %(node)s. "
+                           "Ensure it is installed. Error: %(error)s") %
+                         {'node': task.node.uuid, 'error': e})
+                LOG.error(error_msg)
+                raise exception.RedfishError(error=error_msg)
+            try:
+                reset_job_response = oem_manager.reset_idrac(
+                                                manager=manager)
+            except sushy.exceptions.SushyError as e:
+                LOG.debug("Sushy OEM extension Python package "
+                          "sushy-oem-idrac failed to reset idrac"
+                          "for %(node)s, Error : %(error)s" %
+                          {'node':task.node.uuid, 'error':e})
+        url = oem_manager._conn._url
+        redfish_node_ip = re.search(r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}',
```

Comment:
According to the docs, redfish_address could be a DNS name.  As a result, I think we need to clip out everything between "http[s]://" and the next "/" and use that as the hostname to ping

Reply:
Addressed, used urlparse to clip out hostanme from redfish_address.